pkg/executor/executor.go

```diff
@@ -116,7 +119,42 @@ func (executor *Executor) serveCreateFuncServices() {
 		req := <-executor.requestChan
 		fnMetadata := &req.function.ObjectMeta
 
+		var exactlyOneChan *multicast.Channel[*fscache.FuncSvc]
+
 		if req.function.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType == fv1.ExecutorTypePoolmgr {
+
+			// This is a special case to ensure that if a Function spec has a concurrency of one
+			// we only ever spawn one of those pods.
+			// If we end up in this create function and see that another thread is already specializing
+			// we wait for it to broadcast the result instead of creating a new pod.
+			if req.function.Spec.Concurrency == 1 {
+				maybeChan, loaded := executor.exactlyOneChans.LoadOrStore(req.function.UID, multicast.New[*fscache.FuncSvc]())
```

Comment:
The function UID as a cache key is good, but is there a way to incorporate the current environment version in some way as well? That way, if incoming requests queue up for an old environment, and we deploy, new requests that come in will queue up for the new environment, and won't get served a response that points to a function that's about to die?

Reply:
Oh that's a good point, let me see if we have that info.